Library/Homebrew/version.rb

```diff
@@ -651,7 +651,7 @@ def freeze
 
   protected
 
-  sig { returns(String) }
+  sig { returns(T.nilable(String)) }
```

Comment:
This can never be `nil`. `@version` is only ever set to `val.to_str` in `initialize`.

Reply:
The previous CI run disagreed 😉.

Reply:
Well, it can only ever be `nil` in `NullVersion`, so it should be fixed there.

Reply:
Not sure Sorbet supports that:
```
RuntimeError (Incompatible return type in signature for override of method `version`:)
* Base: `String` (in Version at /opt/homebrew/Library/Homebrew/version.rb:655)
* Override: `T.nilable(String)` (in Version::NullVersion at /opt/homebrew/Library/Homebrew/version/null.rb:101)
(The types must be covariant.)
```

Reply:
This is the Liskov substitution principle IIRC – subtype methods can return a narrower (covariant) type but not something else entirely (`nil`, in this case). (i.e. the signature needs to be fixed in `Version` for type soundness). I don't think I have add'l thoughts here, but happy to answer any questions.